### PR TITLE
refactor: remove `StringOp.Eq/Neq` from `SemanticOperator`

### PR DESCRIPTION
**Pull request by jaschdoc**
Closes https://github.com/flix/flix/issues/6048

### CODE REVIEW

File: main/src/library/Eq.flix [hunk 1]
```diff
@@ -117,8 +117,13 @@ instance Eq[BigInt] {
 }
 
 instance Eq[String] {
-    pub def eq(x: String, y: String): Bool = $STRING_EQ$(x, y)
-    pub override def neq(x: String, y: String): Bool = $STRING_NEQ$(x, y)
+    pub def eq(x: String, y: String): Bool =
+        import java.lang.String.equals(##java.lang.Object): Bool \ {};
+        equals(x, checked_cast(y))
+
+    pub override def neq(x: String, y: String): Bool =
```
Comment (magnus-madsen) [inline main/src/library/Eq.flix:124]: No need to override then? I think we can just leave neq blank :)

**Comment by magnus-madsen** [comment]
> I think `StringOp.Eq` is used for pattern matching somehow in the compiler. I'll investigate further

Hmm

**Comment by magnus-madsen** [comment]
> The problem arises in `Simplifier` but I'm thinking we can generate a call to `Eq.eq` before that

Can you link the code? :)

**Comment by jaschdoc** [comment]
> > The problem arises in `Simplifier` but I'm thinking we can generate a call to `Eq.eq` before that
> 
> Can you link the code? :)

https://github.com/flix/flix/blob/000ca53fec879bf971aebd8db9862064d16d66b7/main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala#L356

**Comment by magnus-madsen** [comment]
That's interesting. We may have to replace that `mkEquals` with something that actually uses ` Eq.eq`.

**Comment by jaschdoc** [comment]
Problem is, I don't see how to emit a function call from the Ast nodes in `SimplifiedAst` (`Eq.eq` has been assigned a symbol during compilation, right?)

**Comment by magnus-madsen** [comment]
You may be right. Can we start by dropping the String.Neq then and see.

Let us ask @mlutze when he is home.

**Comment by mlutze** [comment]
Can we just change it to `InvokeStaticMethod(Object.equals)`?

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**